Laravel 10 compatibility

SimonBarrettACT opened the pull request:
Packages have been updated to be compatible with Laravel 10.

SimonBarrettACT committed:
laravel 10 compatibility

SimonBarrettACT committed:
Update composer.json

Set compatibility with Laravel 10

SimonJnsson commented:
Hey @SimonBarrettACT, thanks for contributing!

Could you please also add Laravel 10 to the github [workflow tests](https://github.com/Morning-Train/data-transfer-object-casters/blob/c6ac3fe538785c8e55516794595b3b766aefc8fd/.github/workflows/run-tests.yml#LL17C10-L17C10), so we can ensure the checks are passing for Laravel 10 aswell?

SimonBarrettACT committed:
Update run-tests.yml

Add Laravel 10

SimonJnsson commented on .github/workflows/run-tests.yml:
```diff
@@ -14,13 +14,15 @@ jobs:
             testbench: 7.*
+          - laravel: 10.*
+            testbench: 8.*
 
```
We should exclude certain combinations of php and laravel. Something like this.

```suggestion
        exclude:
          - laravel: 8.*
            php: 8.2
          - laravel: 10.*
            php: 8.0
```

SimonJnsson commented on .github/workflows/run-tests.yml:
```diff
@@ -14,13 +14,15 @@ jobs:
       matrix:
         os: [ubuntu-latest, windows-latest]
         php: [8.0]
```
Let's make sure we run the checks for 3 latest php versions,

```suggestion
        php: [8.0, 8.1, 8.2]
```

SimonBarrettACT replied:
Sorry for the delay, I have now incorporated your suggestions.

SimonBarrettACT committed:
Test with latest PHP versions

Test with the 3 latest php versions (8.0, 8.1, 8.2) and exclude incompatible combinations of PHP versions and Laravel versions.

SimonJnsson reviewed (approved):
Thanks for contributing!
